client/src/components/__tests__/DynamicJsonForm.test.tsx

```diff
@@ -35,6 +36,130 @@ describe("DynamicJsonForm String Fields", () => {
       expect(input).toHaveProperty("type", "text");
     });
   });
+
+  describe("Format Support", () => {
+    it("should render email input for email format", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        format: "email",
+        description: "Email address",
+      };
+      render(<DynamicJsonForm schema={schema} value="" onChange={jest.fn()} />);
+
+      const input = screen.getByRole("textbox");
+      expect(input).toHaveProperty("type", "email");
+    });
+
+    it("should render url input for uri format", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        format: "uri",
+        description: "Website URL",
+      };
+      render(<DynamicJsonForm schema={schema} value="" onChange={jest.fn()} />);
+
+      const input = screen.getByRole("textbox");
+      expect(input).toHaveProperty("type", "url");
+    });
+
+    it("should render date input for date format", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        format: "date",
+        description: "Birth date",
+      };
+      render(<DynamicJsonForm schema={schema} value="" onChange={jest.fn()} />);
+
+      const input = screen.getByDisplayValue("");
+      expect(input).toHaveProperty("type", "date");
+    });
+
+    it("should render datetime-local input for date-time format", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        format: "date-time",
+        description: "Event datetime",
+      };
+      render(<DynamicJsonForm schema={schema} value="" onChange={jest.fn()} />);
+
+      const input = screen.getByDisplayValue("");
+      expect(input).toHaveProperty("type", "datetime-local");
+    });
+  });
+
+  describe("Enum Support", () => {
+    it("should render select dropdown for enum fields", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        enum: ["option1", "option2", "option3"],
+        description: "Select an option",
+      };
+      render(<DynamicJsonForm schema={schema} value="" onChange={jest.fn()} />);
+
+      const select = screen.getByRole("combobox");
+      expect(select.tagName).toBe("SELECT");
+
+      const options = screen.getAllByRole("option");
+      expect(options).toHaveLength(4);
+    });
+
+    it("should use enumNames for option labels", () => {
+      const schema: JsonSchemaType = {
+        type: "string",
+        enum: ["val1", "val2"],
+        enumNames: ["Label 1", "Label 2"],
```

Comment:
`enumNames` [is not part of the JSONSchema spec](https://github.com/rjsf-team/react-jsonschema-form/issues/532). The [recommended way to describe](https://github.com/json-schema-org/json-schema-spec/issues/57) enum labels is with `oneOf` or `anyOf`.